retrying.py

```diff
@@ -214,7 +210,84 @@ def should_reject(self, attempt):
 
         return reject
 
+    def retry_main_loop(self):
```

Comment:
This generator yield each time it requires an attempt. This let `call_sync` or `call_async` call `fn` approrietly, without worrying the retry marshalling.